### DISCUSSION TITLE
Make default note filename more customizable

### DISCUSSION THREAD

**Discussion by vHanda** [GitJournal/GitJournal#841] [Feature Requests]
The few settings we have right now seem to not be enough.

https://github.com/GitJournal/GitJournal/issues/23#issuecomment-578520591

---
> Please vote on this issue with a 👍reaction. It's one of the primary mechanisms used to prioritize issues.

**Comment by devoldski** [comment]
again thank you for making this useful app.
I suggest that it should be possible to set some simple yet powerful rules for naming conventions. 
would be nice to be able to set default    using date+yaml front matter values in example: "date(Ymd)-yaml(key)"
if no key exist you cold use existing naming convention

**Comment by jrollins** [comment]
Honestly, I don't see the motivation for being able to configure the file name.  I think the file name should always be something unique, like the creation ISO time in UTC or some other unique identifier, and should not be configurable.  Allowing it to be the note title means that the note file name would have to change every time the note title was changed, which seems annoying and fragile to me.  I don't like that the default is that the note title is used as file name, and would prefer the default be a unique identifier that would never have to change over the lifetime of the note.

**Comment by vHanda** [comment]
Hey James. I also prefer having a unique identifier. And actually, I totally forgot about the case when the title changes.

Though given this is maybe the 4th user who has requested such a feature, perhaps we should provide it.

I'm thinking of changing the default back to the current datetime, but instead of the iso format using something simpler, as windows (and Jekyll) have problems with the `:`

**Comment by jrollins** [comment]
> Hey James. I also prefer having a unique identifier. And actually, I totally forgot about the case when the title changes.
> 
> Though given this is maybe the 4th user who has requested such a feature, perhaps we should provide it.
> 
> I'm thinking of changing the default back to the current datetime, but instead of the iso format using something simpler, as windows (and Jekyll) have problems with the `:`

Oh that's too bad.  Maybe the unique ID could be the unix timestamp of the UTC creation time?  (I suggest UTC time because of the time zone ambiguity otherwise.)

I think my main point is that the for best interoperability with external apps (I'm developing a [python interface](https://gitlab.com/jrollins/xotes)) it would be *much* preferable if the note file name was a completely unique identifier that didn't change over the lifetime of the note.
